### PR TITLE
deisctl list-machines

### PR DESCRIPTION
**Pull request by michelleN**

### CODE REVIEW

File: deisctl/deisctl.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ Commands, use "deisctl help <command>" to learn more:
   install           install components, or the entire platform
   uninstall         uninstall components
   list              list installed components
+  list-machines     Enumerate the current hosts in the cluster
```
Comment (sgoings) [inline deisctl/deisctl.go]: Since we're already a bit different between fleetctl and deisctl... would `machines` be a better choice than `list-machines`?

Comment (sgoings) [inline deisctl/deisctl.go]: Additionally... you'll probably want lowercase the command's explanation.

Comment (bacongobbler) [inline deisctl/deisctl.go]: I'd argue for `nodes` as a better choice in prep for v2 (would be handy to make `deisctl nodes` be equivalent to `kubectl get nodes`)

File: deisctl/deisctl.go [hunk 2]
```diff
@@ -95,6 +96,8 @@ Options:
 	switch command {
 	case "list":
 		err = c.List(argv)
+  case "list-machines":
+    err = c.ListMachines(argv)
```
Comment (mboersma) [inline deisctl/deisctl.go]: This needs `gofmt` run on the source files--looks like Jenkins complained about this whitespace difference. :white_circle:

File: deisctl/deisctl.go (hunk 1 above)
Comment (bacongobbler) [inline deisctl/deisctl.go]: We bickered amongst ourselves and found `machines` to be the best candidate for v1. We can change it to `nodes` in v2 if it's too confusing (and as @mboersma pointed out, we can also just make it an alias of `machines`, too).

File: deisctl/client/client.go [hunk 3]
```diff
@@ -240,6 +241,20 @@ Usage:
 	return cmd.ListUnits(c.Backend)
 }
 
+func (c *Client) Machines(argv []string) error {
+	usage := `Enumerate the current hosts in the cluster
```
Comment (krancour) [inline deisctl/client/client.go]: This is an awesome PR, so this is nitpicking mode here.  Does "enumerate" actually imply some ordinality to the results that actually isn't present?  Might it be better to just say, "List the current hosts in the cluster?"

File: deisctl/backend/fleet/list_machines.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+			return ms.PublicIP
+		},
+		"metadata": func(ms *machine.MachineState, full bool) string {
+			if len(ms.Metadata) == 0 {
+				return "-"
+			}
+			return formatMetadata(ms.Metadata)
+		},
+	}
+)
```
Comment (krancour) [inline deisctl/backend/fleet/list_machines.go]: This is clever using a map of functions to handle each field.  Looking at the rest of the code and docstrings tho, I don't see any supported options for limiting what fields are returned or requesting additional fields.  Are there plans to do so?  If not, I'm almost wondering if there's some complexity that we could pare down here by just moving this logic right into `printMachines()`

**Comment by krancour** [comment]
This is an awesome PR and very thoroughly tested.  I imagine the many commits are intended to be squashed into one in this case, but make sure that the final commit(s) follow the project's [commit style](http://docs.deis.io/en/latest/contributing/standards/#commit-style).

File: deisctl/client/client.go (hunk 3 above)
Comment (michelleN) [inline deisctl/client/client.go]: We were just copying the fleetctl list-machines cmd description here, but yes list works too.

Comment (krancour) [inline deisctl/client/client.go]: Makes sense.  We don't need to be just like them, but this is enough precedent to probably leave it be.  :smile:

File: deisctl/backend/fleet/list_machines.go (hunk 4 above)
Comment (adamreese) [inline deisctl/backend/fleet/list_machines.go]: I left it open so we could support other args for full machine ids or limiting columns.  I am not sure how much value we are adding by making a clone of the `fleetctl list-machines` command.

**Pull request merged**